qos/evm/endpoint.go

```diff
@@ -176,3 +124,31 @@ func parseBlockNumberResponse(response string) *uint64 {
 	}
 	return &parsed
 }
+
+// applyArchivalObservation updates the archival check if a valid observation is provided.
+func (e *endpoint) applyArchivalObservation(archivalResponse *qosobservations.EVMGetBalanceResponse) {
+	e.checkArchival = endpointCheckArchival{
+		observedArchivalBalance: archivalResponse.GetBalance(),
+		expiresAt:               time.Now().Add(checkArchivalInterval),
+	}
+}
+
+// getBlockNumber returns the parsed block number value for the endpoint.
+func (e endpoint) getBlockNumber() (uint64, error) {
```

Comment:
This is adding decision making to the endpoint, rather than simply storing the data for other entities to process (granted it needs a refactor already)